Blog/urls.py

```diff
@@ -7,13 +7,13 @@
 urlpatterns = [
     path('', views.index, name="index"),
     path('posts/', views.PostListView.as_view(), name="posts"),
-    path('posts/<int:pk>', views.post_detail_view, name="post-detail"),
+    path('posts/view/<slug:slug>', views.post_detail_view, name="post-detail"),
     path('comments/<int:pk>/add', views.add_comment_form, name="comment-add"),
     path('comments/<int:pk>/do-add', views.add_comment, name="comment-add-do"),
     path('posts/add', views.PostCreate.as_view(), name="post-add"),
     path('vote', views.vote_content_model, name="vote"),
     path('users/', views.UserListView.as_view(), name="users"),
-    path('user/<int:pk>', views.user_detail_view, name="user-detail"),
+    path('user/view/<slug:slug>', views.user_detail_view, name="user-detail"),
```

Comment:
There could be a problem later when adding a new url under `user/...` (For the reason see comment above).

Also I think it would be more consistent, since the posts also have a `view` segment in the url, if you want to view the details to a post.